nemo/core/neural_modules.py

```diff
@@ -57,11 +51,26 @@ class WeightShareTransform(Enum):
 )
 
 
-class NeuralModule(ABC):
-    """Abstract class that every Neural Module must inherit from.
+class NeuralModule(NeuralInterface):
+    """
+        Abstract class that every Neural Module must inherit from.
     """
 
-    def __init__(self):
+    def __init__(self, name=None):
+        # Initialize the inferface.
+        super().__init__(name)
+
+        # Retrieve dictionary of parameters (keys, values) passed to init.
+        self._init_params = self.__extract_init_params()
+
+        # Get object UUID.
+        self._uuid = str(uuid.uuid4())
+
+        # Register module and store the generated name.
+        self._name = self._app_state.register_module(self, name)
```

Comment:
Is there a reason you pass name to NeuralInterface if you overwrite it here?

Reply:
I don't understand why we need AppState if we already have NeuralModuleFactory

Reply:
hmmm, yet another good catch! I was first setting name in NI, then I realized that I need to make it unique, played with different ways the ObjectRegistry might work... seems that I have left the name by an accident. thanks!

Reply:
done